site/src/pages/SearchPage/CourseHitItem.tsx

```diff
@@ -76,18 +76,9 @@ const CourseHitItem: FC<CourseHitItemProps> = (props) => {
               </Badge>
             ))}
           </div>
-          <div>
-            {onSaveCourse && !courseIsSaved && (
-              <IconButton onClick={(e) => onSaveCourse(e)} size={'small'}>
-                <BookmarkBorderIcon />
-              </IconButton>
-            )}
-            {courseIsSaved && (
-              <IconButton onClick={(e) => onUnsaveCourse(e)} size={'small'}>
-                <BookmarkIcon />
-              </IconButton>
-            )}
-          </div>
+          <IconButton onClick={(e) => (courseIsSaved ? onUnsaveCourse(e) : onSaveCourse(e))} size="small">
+            {courseIsSaved ? <BookmarkIcon /> : <BookmarkBorderIcon />}
+          </IconButton>
```

Comment:
this creates an implicit dependence between conditionals, which seems to me like an anti-pattern. It's more concise, but I think it'd be nice to think of it more like a "toggle button" with a state-dependent icon as opposed to a button with state-dependent functionality and a state-dependent icon.

My ideal solution for something like this would be to combine the onunsave/save functions, use that as a single handler, and then render the icon conditionally. This also isolates function logic to their definitions instead of also being changeable both inline and in definitions (if you can swap which handler is being called, now you have to check two handlers to see if one or both are misbehaving)

```suggestion
          <IconButton onClick={toggleSaveCourse} size="small">
            {courseIsSaved ? <BookmarkIcon /> : <BookmarkBorderIcon />}
          </IconButton>
```